controllers/hetznerbaremetalhost_controller.go

```diff
@@ -66,6 +66,17 @@ type HetznerBareMetalHostReconciler struct {
 // Reconcile implements the reconcilement of HetznerBareMetalHost objects.
 func (r *HetznerBareMetalHostReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.Result, reterr error) {
 	log := ctrl.LoggerFrom(ctx)
+
+	start := time.Now()
+	defer func() {
+		// check duration of reconcile. Warn if it took too long.
+		duration := time.Since(start)
+		if duration < 15*time.Second {
```

Comment:
not sure about defer and return, but I think it makes more sense to just log if the duration is too big.

Reply:
so make it the other way around and it is simpler! 

if duration >= 15*time.Second then log

Reply:
This is the "return early" pattern.

imagine there are more `if` conditions needed.

The code would be much more complex:

Without early return:
```
if cond1 {
    if cond2 {
        if cond3 {
            log()
```

Or with early return:

```
if !cond1 {
    return
}
if !cond2 {
    return
}
if !cond3 {
    return
}
log()
```

May main goal is to get it merged. I switched to the syntax  which you prefer.